packages/astro/src/@types/astro.ts

```diff
@@ -1530,8 +1530,42 @@ export interface AstroUserConfig {
 			 * @description
 			 *
 			 * - `"pathanme": The strategy is applied to the pathname of the URLs
+			 * - `"domain"`: SSR only, it enables support for different domains. When a locale is mapped to domain, all the URLs won't have the language prefix.
+			 *    You map `fr` to `fr.example.com`, if you want to have a blog page to look like `fr.example.com/blog` instead of `example.com/fr/blog`.
+			 *    The localised folders be must in the `src/pages/` folder.
 			 */
 			strategy: 'pathname';
+
+			/**
+			 * @docs
+			 * @name experimental.i18n.domains
+			 * @type {Record<string, string> }
+			 * @default '{}'
+			 * @version 4.3.0
+			 * @description
+			 *
+			 * Configures the URL pattern of one or more supported languages to use a domain (or sub-domain). When a locale is mapped to a domain, a `/[locale]/` path prefix will not be used.
+			 *
+			 *  For example, you can configure the `fr` URLs to be of the form `https://fr.example.com/blog`. If not configured in `domains`, the `defaultLocale` will default to `https://example.com/blog` and any other locale not configured will default to `https://example.com/[locale]/blog`.
+			 *
+			 * ```js
+			 * export default defineConfig({
+			 * 	output: "server",
+			 * 	site: "https://example.com",
+			 * 	i18n: {
+			 * 		defaultLocale: "en",
+			 * 		locales: ["en", "fr", "pt-br", "es"],
+			 *		domains: {
+			 *			fr: "https://fr.example.com",
+			 *    },
+			 * 		routing: {
+			 * 			strategy: "domains"
+			 *		}
+			 *  }
+			 * })
+			 * ```
```

Comment:
``````suggestion
			 * Configures the URL pattern of one or more supported languages to use a custom domain (or sub-domain). When a locale is mapped to a domain, a `/[locale]/` path prefix will not be used. However, localized folders within `src/pages/` are still required, including for your configured `defaultLocale`.
			 *
			 *  Any other locale not configured will default to a localized path-based URL according to your `prefixDefaultLocale` strategy (e.g. `https://example.com/[locale]/blog`).
			 *
			 *
			 * ```js
			 * //astro.config.mjs
			 * export default defineConfig({
			 * 	site: "https://example.com",
			 * 	output: "server", // required, with no prerendered pages
			 *   adapter: node({
			 *     mode: 'standalone',
			 *   }),
			 * 	i18n: {
			 * 		defaultLocale: "en",
			 * 		locales: ["en", "fr", "pt-br", "es"],
			 *     prefixDefaultLocale: false,
			 *		  domains: {
			 *			  fr: "https://fr.example.com",
			 *       es: "https://example.es"
			 *    },
			 *  experimental: {
			 *     i18nDomains: true
			 *  }
			 * })
			 * ```
			 *  
		   * Both page routes built and URLs returned by the i18n helper functions `getAbsoluteLocaleUrl()` and `getAbsoluteLocaleUrlList()` will use the options set in `i18n.domains`.
		   *  
		   * See the [Internationalization Guide](https://docs.astro.build/en/guides/internationalization/#domains-experimental) for more details, including limitations of this experimental feature.
``````